src/ipc/TypedMsgHdr.h

```diff
@@ -57,6 +59,8 @@ class TypedMsgHdr: public msghdr
     /* access to message parts for selected commonly-used part types */
     void getString(String &s) const; ///< load variable-length string
     void putString(const String &s); ///< store variable-length string
+    void getString(SBuf &) const; ///< load variable-length string
```

Comment:
This is being kept identical to `getString(String)` to reduce refactoring diff of future String removals.

Reply:
How would implementing the requested change increase "refactoring diff of future String removals"? I may continue to insist on this change even if it does increase some future diffs (because the change itself is highly desirable and in-scope), but I currently do not see where those alleged diff increases would come from.

Reply:
Changing the API of `TypedMsgHdr::getString()` would mean all PRs changing other members of `Mgr::ActionParams` class, `Mgr::StringParam`, `Ipc::StrandSearchRequest`, and `Ipc::StrandCoord` to change their unpack logic.

With the non-change this PR does the relevant `String` member can be swapped to `SBuf` without touching the unpack logic.

Reply:
> Changing the API of `TypedMsgHdr::getString()` would mean all PRs changing other members of `Mgr::ActionParams` class, `Mgr::StringParam`, `Ipc::StrandSearchRequest`, and `Ipc::StrandCoord` to change their unpack logic.

I now see what you you mean by "reduce refactoring diff of future String removals", thank you.

The unpacking _logic_ in those classes will stay the same (and might even improve in some cases), but the _diff_ will indeed be larger compared to the diff with the wrong/current API. In the future, the suggested API fix may also help avoid creating empty SBuf fields just to immediately fill them with unpacked SBufs -- a waste of quite a few CPU cycles and debugging cache.log lines, but that improvement requires more non-trivial changes.

I still recommend implementing getString(void) API here (instead of reducing the future diff), but I do not insist on this. Your call.

Reply:
How about adding a SBuf getSBuf(); API besides this, to allow replacing callers piecemeal?